data-prepper-expression/src/main/java/org/opensearch/dataprepper/expression/ConditionalExpressionEvaluator.java

```diff
@@ -27,8 +27,20 @@ public ConditionalExpressionEvaluator(final Parser<ParseTree> parser, final Eval
      * <b>Method not implemented</b>
      */
     @Override
-    public Boolean evaluate(final String statement, final Event context) throws ClassCastException {
-        // TODO: Implement method
-        throw new RuntimeException("Method not implemented");
+    public Boolean evaluate(final String statement, final Event context) throws ExpressionEvaluationException {
+        try {
+            final ParseTree parseTree = parser.parse(statement);
```

Comment:
We should cache these `ParseTree` objects so that we do not need to re-parse the statement. Can you create an issue on it so that we can track it?

Reply:
That works even better.